### ISSUE TITLE
Webview in Android does not respect the dark mode

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#4527]
An appropriate dark theme should be applied for all Webviews in Organic Maps, using the simplest approach with as few lines of code as possible:

1. FAQ
2. Copyright
3. Wikipedia article
4. Bookmark description
5. ...something else where Webview is used?

|Wiki|Place Page Description|HTML to reproduce Place Page issue|
|---|---|---|
|![image](https://user-images.githubusercontent.com/170263/219901915-5447f65a-f3ee-4447-89d2-4af1ea240b8f.png)|![image](https://user-images.githubusercontent.com/170263/219901926-fd7c73d3-1e4c-4d5d-990a-540626bdd55d.png)|![image](https://user-images.githubusercontent.com/170263/219901959-6c50cee0-7f15-43bc-92df-a9fffbc494d5.png)|

**Comment by harmanpreet7916** [comment]
I want to contribute to this issue but I am new to this organization and still understanding the structure of the project. I can not find the files where i can edit the code to contribute to this issue can someone help me with this. i have read  docs>styles.md but still not getting how to get started.

**Comment by biodranik** [comment]
Please check all documentation in the docs folder. One of the important files is docs/INSTALL.md

**Comment by satvik2131** [comment]
I guess this issue is solved , am I right ? I have checked the FAQ section and it is working fine 
Device : Pixel 4a

**Comment by Jean-BaptisteC** [comment]
https://developer.android.com/develop/ui/views/layout/webapps/dark-theme

**Comment by satvik2131** [comment]
@Jean-BaptisteC should we use Algorithmic Darkening ? is it good ?

**Comment by Jean-BaptisteC** [comment]
Yes, i guess (i'm not an expert in theming), actual themes in OM can work whit Algoritmic Darkening ?

**Comment by satvik2131** [comment]
> Yes, i guess (i'm not an expert in theming), actual themes in OM can work whit Algoritmic Darkening ?

I have to try to answer this , I will give it a try if it works fine I will let you know

**Comment by euf** [comment]
Linking #3042 as iOS has similar problem with wikipedia webview in dark mode

**Comment by euf** [comment]
As a feasible solution for both Android and iOS, I would suggest preparing two CSS files (white and dark) and then injecting them in the runtime for all HTML content depending on current app theme. It would work for all cases in the original issue (FAQ, copyright, Wikipedia articles, etc). 

iOS reference: https://stackoverflow.com/questions/61893390/how-to-load-css-file-by-using-url-in-wkwebview

Android reference: https://stackoverflow.com/questions/67754291/is-there-a-way-to-have-an-webview-on-android-where-i-can-inject-css-before-the-p

**Comment by Jean-BaptisteC** [comment]
@biodranik do you think is the best approach, use CSS files to fix theming in webview ?

**Comment by biodranik** [comment]
The code is custom/different anyway on Android and iOS, so that is not the point in choosing a "universal" approach. I would do it in the simplest/most robust/battery-friendly way.

I would follow the official approach for each platform, which will work consistently on older and newer OS versions, and will cover all our review-related cases.

**Comment by Jean-BaptisteC** [comment]
I have done some searches:
- FAQ and Copyright screen respect dark mode when dark mode is enabled in Android settings.
- These 2 app screens used custom WebContainerDelegate

I think the solution is to use everywhere WebContainerDelegate and do a real dark mode recognized by Android.

**Comment by ngosang** [comment]
I can reproduce this issue in the release 2024.05.11-3. Dark theme works everywhere but in the full Wikipedia article.

![Screenshot_2024-05-28-19-16-47-04_75d0290a7c00525acd83c0ab8700d64c](https://github.com/organicmaps/organicmaps/assets/10577978/30d21ee4-7312-4742-ab12-238219bd06cb)
![Screenshot_2024-05-28-19-16-52-71_75d0290a7c00525acd83c0ab8700d64c](https://github.com/organicmaps/organicmaps/assets/10577978/877304e3-7a4f-4c98-9c26-501c594dca39)

**Comment by RedAuburn** [comment]
I got something working for the description, but it's too hacky IMO: https://github.com/RedAuburn/organicmaps/tree/dark-webview

A proper fix will likely be to use the custom WebContainerDelegate, then inject the CSS in that.
